hilti/include/rt/types/bool.h

```diff
@@ -6,8 +6,20 @@
 
 namespace hilti::rt {
 
+class Bool {
```

Comment:
Just for my C++ education: I suppose there's no strong typedef for builtin types, so that we wouldn't need to create a class?

Reply:
AFAIK as of C++17 there is no strong typedef feature in the core language. Since we are wrapping an integral type we could use the following though (which I missed in an almost mechanical way):
```.cpp
enum Bool : bool {
  True = true,
  False = false,
};
```

That would not only fix the issue with `std::vector<bool>`, but as an added benefit introduce literals `True` and `False` (we could make them scoped by using an `enum class`). I'll do that instead.

Reply:
As I suspected, if we'd use an enum type, we'd need to manually convert `Bool` to `bool` when used e.g., in conditionals as conversion functions need to be member functions which we cannot add to an enum. I'd suggest we use the class form for now.